src/NuGet.Core/NuGet.Packaging/Rules/InvalidUndottedFrameworkRule.cs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    internal class InvalidUndottedFrameworkRule : IPackageRule
+    {
+        private const string TargetFramework = "targetFramework";
+
+        private const string Metadata = "metadata";
+
+        private const string Dependencies = "dependencies";
+
+        private const string Group = "group";
+
+        private const string References = "references";
+
+        private const string FrameworkAssemblies = "frameworkAssemblies";
+
+        private const string FrameworkAssembly = "frameworkAssembly";
+
+        private static readonly char[] CommaArray = new char[] { ',' };
+
+        public string MessageFormat { get; }
+
+        public InvalidUndottedFrameworkRule(string messageFormat)
+        {
+            // IPackageRule expects to get messages passed in, but we output
+            // different kinds of messages, so each individual validator is
+            // responsible for its own messaging.
+            MessageFormat = messageFormat;
```

Comment:
Why not remove the parameter from the constructor?

[There are already 3 rules that use the default constructor](https://github.com/NuGet/NuGet.Client/blob/16be25216b699f48d5e4d0baed86a62c78acadbe/src/NuGet.Core/NuGet.Packaging/Rules/RuleSet.cs#L29-L31)